src/languageFeatures.ts

```diff
@@ -120,7 +123,16 @@ export class LibFiles {
 			return model;
 		}
 		if (this.isLibFile(uri) && this._hasFetchedLibFiles) {
-			return editor.createModel(this._libFiles[uri.path.slice(1)], 'typescript', uri);
+			return editor.createModel(this._libFiles[uri.path.slice(1)], this.modeId, uri);
```

Comment:
I would not risk using here `'javascript'` again because that would cause https://github.com/microsoft/monaco-editor/issues/2182 to happen again. My understanding was that the ts services really don't like to see a javascript file created with contents of standard lib `.d.ts` files.